crates/yas_core/src/tests/utils/swap_helper.cairo

```diff
@@ -0,0 +1,412 @@
+            .create_pool(token_0.contract_address, token_1.contract_address, fee_amount);
+        let yas_pool = IYASPoolDispatcher { contract_address: yas_pool_address };
+
+        set_contract_address(OWNER());
+        yas_pool.initialize(initial_price);
+
+        set_contract_address(WALLET());
+
+        mint_positions(yas_router, yas_pool_address, mint_positions);
+
+        (yas_pool, yas_router, token_0, token_1)
+    }
+
+    fn mint_positions(
+        yas_router: IYASRouterDispatcher,
+        yas_pool_address: ContractAddress,
+        mint_positions: @Array<SwapTestHelper::Position>
+    ) {
+        let mut i = 0;
+        loop {
+            if i == mint_positions.len() {
+                break;
+            }
+            let position = *mint_positions[i];
+            yas_router
+                .mint(
+                    yas_pool_address,
+                    WALLET(),
+                    position.tick_lower,
+                    position.tick_upper,
+                    position.liquidity
+                );
+            i += 1;
+        };
+    }
+}
+
+mod ContractDeployerHelper {
+    use yas_core::contracts::yas_erc20::{ERC20, IERC20Dispatcher, IERC20DispatcherTrait};
+    use yas_core::contracts::yas_factory::{
+        YASFactory, IYASFactoryDispatcher, IYASFactoryDispatcherTrait
+    };
+    use yas_core::contracts::yas_pool::{YASPool, IYASPoolDispatcher, IYASPoolDispatcherTrait};
+    use yas_core::contracts::yas_router::{
+        YASRouter, IYASRouterDispatcher, IYASRouterDispatcherTrait
+    };
+
+    use starknet::{ContractAddress, ClassHash, SyscallResultTrait};
+    use starknet::syscalls::deploy_syscall;
+
+    fn deploy_yas_erc20(
+        name: felt252, symbol: felt252, initial_supply: u256, recipent: ContractAddress
+    ) -> IERC20Dispatcher {
+        let mut calldata = array![name, symbol];
+        Serde::serialize(@initial_supply, ref calldata);
+        calldata.append(recipent.into());
+
+        let (address, _) = deploy_syscall(
+            ERC20::TEST_CLASS_HASH.try_into().unwrap(), 0, calldata.span(), true
+        )
+            .unwrap_syscall();
+
+        return IERC20Dispatcher { contract_address: address };
+    }
+
+    fn deploy_yas_router() -> IYASRouterDispatcher {
+        let (address, _) = deploy_syscall(
+            YASRouter::TEST_CLASS_HASH.try_into().unwrap(), 0, array![].span(), true
+        )
+            .unwrap_syscall();
+
+        return IYASRouterDispatcher { contract_address: address };
+    }
+
+    fn deploy_yas_factory(
+        deployer: ContractAddress, pool_class_hash: ClassHash
+    ) -> IYASFactoryDispatcher {
+        let (address, _) = deploy_syscall(
+            YASFactory::TEST_CLASS_HASH.try_into().unwrap(),
+            0,
+            array![deployer.into(), pool_class_hash.into()].span(),
+            true
+        )
+            .unwrap_syscall();
+
+        return IYASFactoryDispatcher { contract_address: address };
+    }
+}
```

Comment:
The idea is that other functions can use this. I didn't refactor it in this PR because it's not within the scope. However, I did this with the thought that it can be reused from here.